### DISCUSSION TITLE
application for join Project "Connecting to the Python Molecular Dynamics community" in GSoC 2023

### DISCUSSION THREAD

**Discussion by Clarkszw** [d-SEAMS/seams-core#11] [GSoC Discussions]
Hi Rohit Goswami and Amrita Goswami,

My name is Zhiwen Shi and you can call me Clark. I am so excited to find seams-core in GSoC 2023 and can’t wait to talk to you about the project and possibilities we could do in this summer.

Let me give a brief about myself: I am now living in Denmark with my wife and two little kids. I graduated from the University of Copenhagen as a master of computational chemistry. I have a strong interest in Cloud Technology and the implementation of HPC with could solutions. Among these solutions, there are many new and popular python based scientific packages involved, fx OpenMM, PySCF, AiiDA. These open source packages motivate me to get know more about the software engineering of a python package. I am seeking a good start up project for myself to get on board.

I think seams-core with GSoC will be a great match for me: 1. Python related. 2. New contributor friendly. 3. With Mentor. 4. Chemistry related. 5. conda related, 6. related with AiiDA and LAMMPS.

I have read the ‘expectations’ page, here is my short answer for them:

40 hours/week => I can take this more than a part time project to achieve more. So I have the commitment to it.
Self-motivation and steady schedule => I am self-motivated and I am good at it. I have two kids to take care, so I am serious about how to use my time. This is what I love to do. I will spend good time with mentor to make a good plan for the whole project and break the plan down to every week, everyday. We will together make the most out of it :)
Frequent communication with mentor => We will have a good relationship during the project. The mentor is the leader and I am the runner. Teamwork works. We will also make up a plan about the communication, how often, via what, how about the regular communication, how about the urgent? etc..
Version control => I am new to a big "hands on" project, but I am not new to use ‘git’. I am aware of the importance of this tool in the project.
Community => I’d love to join the community of seams-core and I think it is spirit of the open source project. I will ask a lot and hopefully answer a lot in the near future.
Design documents => I will work out this with my mentor and maintain it. It will be a memorable documentation of the project.

Regarding the project that I am interested, I would like to start with "Connecting to the Python Molecular Dynamics community" with mentors @HaoZeke  and @amritagos. I hope I can hear from you soon :)

I am currently going through the readme file on your GitHub page and would love to hear of any tips to get started! Thanks in advance.

That’s all I want to say to express my excitement. Have a lovely week :)

Best Wishes
Zhiwen (Clark)

**Comment by HaoZeke** [comment]
Hi Clark.

Thanks for reaching out! It sounds like you'd be a great fit for the "Connecting to the Python Molecular Dynamics community" project. As we are participating under the PSF, there are some additional requirements enumerated here: https://python-gsoc.org/index.html#gettingstarted

One of the things I would suggest, is trying to setup the development version of the code, and for example, trying to run all the examples, if there are any missing paths or changed files, that would make a great first contribution.

Aside from that, students are expected to have a blog, so even a blog post (say with a static site like Hugo [with Netlify](https://www.netlify.com/blog/2016/10/27/a-step-by-step-guide-deploying-a-static-site-or-single-page-app/)) could be a great start.

I am happy to see you've gone over the requirements and will be able to commit to the time-constraints and hope we have an effective collaboration if all goes well as well.

Either after you have taken an attempt at this or otherwise, feel free to share a draft of your application as well, we (@amritagos and I) would be happy to provide inputs as well.

**Reply by HaoZeke** [reply]
Additionally, we could also setup a Zoom call over email sometime this week, my email is rog32@hi.is

I'm in the process of working with some of the other applicants as well, who are less Github savvy and needed some setup assistance for the other project.

**Reply by Clarkszw** [reply]
Hi @HaoZeke , 
Thanks for the reply. 
I am in Denmark, now is 15:17. I don't know if will have time to have a short talk in the rest of day. Or if you work also in the weekend?
I will go through the readme file, setup the development version of code and run all the examples.
If these go well, I will drafting my application and send you before I submit to GSoC.

**Reply by HaoZeke** [reply]
Sorry I was a bit tied up with other things today, I sent details to chat over the weekend :)
